js/stdlib/ts/ALGO.ts

```diff
@@ -1671,40 +1671,73 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
         const seek = async (t: Time) => {
           debug("EventStream::seek", t);
           time = t;
-          logIndex = 0;
+          // logIndex = 0;
         }
 
         const next = async () => {
           let dhead = "EventStream::next";
           const { ApplicationID, getLastRound } = await getC();
           if (time == null) {
             time = bigNumberify(getLastRound());
+            logIndex[time.toNumber()] = 0;
           }
           const pred = (txn: any) => {
-            const logs: string[] = (txn['logs'] || []).slice(logIndex);
+            const round = txn['confirmed-round'];
+            const logIdx = logIndex[round] || 0;
+            const logs: string[] = (txn['logs'] || []).slice(logIdx);
             const good = logs.some((log) => log.startsWith(base64Hash));
             return good;
           };
-          const res = await eventCache.query(dhead, ApplicationID, { minRound: time.toNumber() }, pred);
-          if (!res.succ) {
-            throw Error(`Event not found`);
+          let res: QueryResult = { succ: false, round: 0  };
+          while (!res.succ) {
+            res = await eventCache.query(dhead, ApplicationID, { minRound: time.toNumber() }, true, pred);
```

Comment:
I'm not sure what `busy way` means

Reply:
I meant "busy wait"

Reply:
This may be occurring. When I put a `wait` in the example contracts, docker uses ~200% CPU, as opposed to the normal 150% it uses. What would be a technique to avoid busy wait? Putting a sleep timer?

Reply:
Yea, using Timeout. You know that a new block can't happen in less than like 5 seconds, so you can wait 5s